src/typhoon_automator/simulation.py

```diff
@@ -39,22 +52,72 @@ def schedule_event(
     def invoke_event(
             self,
             event: Any):
-        raise NotImplementedError()
+        if event is None:
+            raise ValueError("Event cannot be None")
+        try:
+            event_time = round(self.get_simulation_time(), 6)
+            self._automator.log(f"Event at {event_time}: {event.message}")
+            event.invoke(self)
+      
+        except AttributeError:
+          if not hasattr(event, "invoke"):
+            self._automator.log("Event object does not have an invoke method", level = logging.ERROR) # is this the right level?
```

Comment:
`Error` is good!  It's a problem, so it's more severe than `Warning`, but the simulation _can_ continue, so it's not 'crash-and-burn' enough to be `Critical`.

Fun fact: this is related to the discussion we had about Python not having interfaces, meaning we have to write this code defensively.  Specifically, we use `event` expecting it to have an `invoke` method and a `message` string, and `AttributeError` gets thrown when it doesn't.